src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -5804,6 +5826,74 @@ public enum EventManifestOptions
         AllowEventSourceOverride = 0x8,
     }
 
+#if FEATURE_EVENTSOURCE_XPLAT 
+
+    internal  class XplatEventLogger : EventListener
+    {
+        public XplatEventLogger() {}
+        
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        public static extern bool IsEventSourceLoggingEnabled();
+
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern void LogEventSource(int eventID, string eventName, string eventSourceName, string payload);
+
+        private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnlyCollection<object> payload, string sep = ", ")
+        {
+
+            var sb = StringBuilderCache.Acquire();
+
+            if (payloadName == null || payload == null )
+                return String.Empty;
+
+            if (payloadName.Count == 0 || payload.Count == 0)
+                return String.Empty;
+
+            Contract.Assert(payloadName.Count == payload.Count);
+
+            for (int i = 0; i < payloadName.Count; i++)
+            {
+                var fieldstr = payloadName[i].ToString();
+                var valuestr = payload[i].ToString();
+
+                sb.Append(fieldstr);
+                sb.Append(":");
+                sb.Append(valuestr);
+                sb.Append(sep);
+
+            }
+
+             sb.Remove(sb.Length - sep.Length, sep.Length);
+             return StringBuilderCache.GetStringAndRelease(sb);
+
+        }
+
+        internal protected  override void OnEventSourceCreated(EventSource eventSource)
+        {
+            EnableEvents(eventSource, EventLevel.LogAlways, EventKeywords.All, null);
+        }
+
+        internal protected  override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            LogOnEventWritten(eventData);
+        }
+
+        [System.Security.SecuritySafeCritical]
+        private void LogOnEventWritten(EventWrittenEventArgs eventData)
+        {
+            string payload = "";
+            if (eventData.Payload != null)
+            {
+                payload = Serialize(eventData.PayloadNames, eventData.Payload);
+            }
+
+            LogEventSource( eventData.EventId, eventData.EventName,eventData.EventSource.Name,payload);
```

Comment:
This design is all or nothing.  However, one option that @vancem suggested, and I passed on to @ramarag was to partition the most common EventSources into their own LTTng events.  E.g. TPL, FrameworkEventSource, etc.  This would get us some separation until we can implement on-the-fly event registration, which is something that LTTng doesn't support today.

My belief is that separating out a handful of the most common EventSources will be a good way to make the trade-off between complexity and runtime performance cost.